radareorg/r2ghidra-dec
# Function name annotation annotator implemented

**Detailed description**
This PR implements an annotator that will annotate function names with a new type of annotation (`R_CODE_ANNOTATION_TYPE_FUNCTION_NAME`) introduced by [PR #17204](https://github.com/radareorg/radare2/pull/17204) in radare2. This will be required for implementing some features in the decompiler widget in Cutter.

I have also implemented [offset annotation](https://github.com/radareorg/r2ghidra-dec/pull/123#discussion_r451417915) for the currently decompiled function's name in this PR.



...

**Test plan**
1. Fetch [PR #17204](https://github.com/radareorg/radare2/pull/17204) in radare2 and compile it before compiling this.
2. Make sure the output is working as expected from the JSON output (check r2 PR first for making sure JSON output is implemented correctly for function names).
3. Check code and other things that should be checked.


...

**Closing issues**



...
